flink-connectors/flink-connector-pulsar/src/test/java/org/apache/flink/connector/pulsar/sink/writer/PulsarWriterTest.java

```diff
@@ -44,32 +46,182 @@
+        operator().createTopic(topic, 8);
+
+        try (final PulsarWriter<String> writer = createWriter(topic, initContext)) {
+            String message = randomAlphabetic(10);
+            // No producer is present, currentSendTime should be INVALID_LAST_SEND_TIME
+            assertThat(metricListener.getGauge(CURRENT_SEND_TIME_METRIC_NAME)).isPresent();
+            Long sendTimeBeforeProducerCreation =
+                    metricListener.<Long>getGauge(CURRENT_SEND_TIME_METRIC_NAME).get().getValue();
+            assertThat(sendTimeBeforeProducerCreation)
+                    .isEqualTo(PulsarSinkWriterMetrics.INVALID_LAST_SEND_TIME);
+
+            // Write a message
+            writer.write(message, CONTEXT);
+            // Advance Timer to reflect the time change after flush
+            long definedSendTime = 50;
+            timeService.advance(definedSendTime);
+            writer.flush(false);
+
+            // After send success, send time should be updated
+            Long sendTimeAfterProducerCreation =
+                    metricListener.<Long>getGauge(CURRENT_SEND_TIME_METRIC_NAME).get().getValue();
+            assertThat(sendTimeAfterProducerCreation).isEqualTo(definedSendTime);
+        }
+    }
+
+    @Test
+    void numAcksReceived() throws Exception {
+        String topic = randomAlphabetic(10);
+        operator().createTopic(topic, 8);
+
+        try (final PulsarWriter<String> writer = createWriter(topic, initContext)) {
+            String message = randomAlphabetic(10);
+
+            // Write and flush a  message
+            writer.write(message, CONTEXT);
+            writer.flush(false);
+            // Advance Timer to reflect the time change after flush
+            sleepUninterruptibly(TEST_PULSAR_STATS_INTERVAL_SECONDS + 1, TimeUnit.SECONDS);
+            timeService.advance(
+                    TimeUnit.SECONDS.toMillis(TEST_PULSAR_STATS_INTERVAL_SECONDS) + 100);
+
+            Long numAcks =
+                    metricListener.<Long>getCounter(NUM_ACKS_RECEIVED_METRIC_NAME).get().getCount();
+            assertThat(numAcks).isEqualTo(1);
+        }
+    }
+
+    @Test
+    void standardMetricsFromIOMetricsGroup() throws Exception {
+        String topic = randomAlphabetic(10);
+        operator().createTopic(topic, 8);
+
+        try (final PulsarWriter<String> writer = createWriter(topic, initContext)) {
+            String message = randomAlphabetic(10);
+            Counter recordsOutCounter =
+                    initContext.metricGroup().getIOMetricGroup().getNumRecordsOutCounter();
+            Counter bytesOutCounter =
+                    initContext.metricGroup().getIOMetricGroup().getNumBytesOutCounter();
+            // Write and flush a  message
+            writer.write(message, CONTEXT);
+            writer.flush(false);
+            // Advance Timer to update counter metrics
+            sleepUninterruptibly(TEST_PULSAR_STATS_INTERVAL_SECONDS + 1, TimeUnit.SECONDS);
```

Comment:
Here we need to sleep because Pulsar only updates `totalXXX` every 60s by default, we need to wait for pulsar producer to have `totalXXX` updated to reflect the changes